### PR TITLE
feat(content-distribution): handle post thumbnail

### PR DESCRIPTION
**Pull request by miguelpeixe**
Add support for syncing the post thumbnail when distributing content.

### Testing

1. Make sure you have a network setup with at least 1 hub and 1 node
2. On the hub, create a new post with a featured image, grab the post ID, and use the WP shell to distribute to the node:

```php
$post_id = 0; // Change to your post ID.
$site_urls = [ 'http://node1.local' ]; // Change to match your setup urls.
Newspack_Network\Content_Distribution::set_post_distribution( $post_id, $site_urls );
Newspack_Network\Content_Distribution::distribute_post( $post_id );
```

3. Visit the media library and confirm the post thumbnail was created
4. Confirm the post is added as a draft and the thumbnail is assigned
5. In the original post, change the featured image and save
6. Visit the media library in node1 and confirm that the previous featured image has been deleted and the new one was created
7. Confirm the new featured image was also updated in the post
8. In the original post, delete the featured image and save
9. Back in node1, confirm in the media library that the featured image has been deleted
10. Confirm in the linked post that no featured image is assigned

### CODE REVIEW

File: includes/class-content-distribution.php [hunk 1]
```diff
@@ -338,6 +339,44 @@ protected static function is_post_unlinked( $post_id ) {
...
+			$post_payload &&
+			$post_payload['post_data']['thumbnail_url'] === $thumbnail_url
+		) {
+			return;
+		}
+
+		// Delete existing thumbnail attachment.
+		if ( $current_thumbnail_id ) {
+			wp_delete_attachment( $current_thumbnail_id, true );
+		}
```
Comment (miguelpeixe) [inline includes/class-content-distribution.php]: When syncing the post thumbnail, whether the distributed post is updating the featured image or deleting it, the destination site will permanently delete the orphan image.

I'm not sure if this is too aggressive, but I also don't like the idea of lingering orphan images.

The current approach can be problematic if:

- The site starts using the synced attachment in other content
- The site unlinked the post, changed the featured image to something local, and re-linked the post - deleting their local attachment

We can prevent the last one above by adding a meta to the synced attachment so only those get deleted.

Comment (naxoc) [reply includes/class-content-distribution.php]: It's probably a pretty expensive query, but do you think it's worth checking if other posts are using the image before deleting it?

There is also the use case that the image is used in the post body.

File: includes/class-content-distribution.php [hunk 2]
```diff
@@ -339,6 +339,44 @@ protected static function is_post_unlinked( $post_id ) {
...
+
+		if ( ! function_exists( 'media_sideload_image' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/media.php';
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+			require_once ABSPATH . 'wp-admin/includes/image.php';
+		}
+
+		$attachment_id = media_sideload_image( $thumbnail_url, $post_id, '', 'id' );
+		if ( is_wp_error( $attachment_id ) ) {
+			return;
```
Comment (naxoc) [inline includes/class-content-distribution.php:375]: A silent error here is probably fine, so this is a question more than a request – but do we have a way to log stuff like this?

**Review by naxoc — COMMENTED** [review]
This looks good and tests well. (The unit-test is broken, but that is from the updated-date-PR I think).

I added some comments about the image deletion though. I think there are use cases we can get in trouble with.

File: includes/class-content-distribution.php (hunk 1 above)
Comment (miguelpeixe) [reply includes/class-content-distribution.php]: I've decided to not delete the attachment for now. Instead, we set a post meta to not lose track of the attachments that came via distribution and we can decide later how to deal with them. WDYT? e5d14189d3ba9e60f0bbde88bebe892d92803f0d

I think a post content query is not viable, even if executed via cron. It's very expensive...

Comment (naxoc) [reply includes/class-content-distribution.php]: I think that makes sense. I like the idea with the postmeta. Better have a bit too much media than media missing.

File: includes/class-content-distribution.php (hunk 2 above)
Comment (miguelpeixe) [reply includes/class-content-distribution.php:375]: Good call! We have a Debugger class, added a log in 23b93e649d39e0091f7b28eaabd1b6b7f6ab89c5

**Comment by miguelpeixe** [comment]
Thank you for the review, @naxoc!

I've addressed the comments and fixed the tests in 9c7212b648285930ec9f3ea787bceb2a653f6482

**Review by naxoc — APPROVED** [review]
Good to go!
ok

**Pull request merged**